internal/graph/builder_test.go

```diff
@@ -0,0 +1,122 @@
+package graph
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/internal/mocks"
+)
+
+func TestNewLayeredCheckResolver(t *testing.T) {
+	type Test struct {
+		name                      string
+		cacheEnabled              bool
+		dispatchThrottlingEnabled bool
+		expectedResolverOrder     []CheckResolver
+	}
+
+	tests := []Test{
+		{
+			name:                      "when_nothing_is_enabled",
+			cacheEnabled:              false,
+			dispatchThrottlingEnabled: false,
+			expectedResolverOrder:     []CheckResolver{&CycleDetectionCheckResolver{}, &LocalChecker{}},
+		},
+		{
+			name:                      "when_cache_alone_is_enabled",
+			cacheEnabled:              true,
+			dispatchThrottlingEnabled: false,
+			expectedResolverOrder:     []CheckResolver{&CycleDetectionCheckResolver{}, &CachedCheckResolver{}, &LocalChecker{}},
+		},
+		{
+			name:                      "when_dispatch_throttling_alone_is_enabled",
+			cacheEnabled:              false,
+			dispatchThrottlingEnabled: true,
+			expectedResolverOrder:     []CheckResolver{&CycleDetectionCheckResolver{}, &DispatchThrottlingCheckResolver{}, &LocalChecker{}},
+		},
+		{
+			name:                      "when_both_are_enabled",
+			cacheEnabled:              true,
+			dispatchThrottlingEnabled: true,
+			expectedResolverOrder:     []CheckResolver{&CycleDetectionCheckResolver{}, &CachedCheckResolver{}, &DispatchThrottlingCheckResolver{}, &LocalChecker{}},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			opts := []CheckQueryBuilderOpt{}
+			if test.cacheEnabled {
+				opts = append(opts, WithCachedCheckResolver())
+			}
+			if test.dispatchThrottlingEnabled {
+				opts = append(opts, WithDispatchThrottlingCheckResolver())
+			}
+			opts = append(opts, WithLocalChecker())
```

Comment:
The fact that you had to write this line for every test makes me think that this should be made by default in the actual code. I can't think of a scenario today where the LocalChecker would _not_ be wanted.

Reply:
It's been sometime but I remember implementing it that way and feeling it wasn't explicit enough. It was abstracted and didn't give the full picture of the builder

Reply:
🤔 to whom should it be explicit? you can say in the godoc of the builder that the last element in the chain is always LocalChecker

Reply:
We are keeping this for the consistency of the builder methods